src/Lachain.Core/Blockchain/SystemContracts/NativeTokenContract.cs

```diff
@@ -22,10 +24,27 @@ public class NativeTokenContract : ISystemContract
         private static readonly ILogger<NativeTokenContract> Logger =
             LoggerFactory.GetLoggerForClass<NativeTokenContract>();
 
+        private static readonly ICrypto Crypto = CryptoProvider.GetCrypto();
+
         private readonly InvocationContext _context;
 
         private readonly StorageMapping _allowance;
 
+        private static EcdsaKeyPair _minterKeyPair =
+            new EcdsaKeyPair("0xD95D6DB65F3E2223703C5D8E205D98E3E6B470F067B0F94F6C6BF73D4301CE48".HexToBytes()
+                .ToPrivateKey());
+        private static byte[] _minterPubKey = CryptoUtils.EncodeCompressed(_minterKeyPair.PublicKey);
+        private static UInt160 _minterAdd = Crypto.ComputeAddress(_minterPubKey).ToUInt160();
+
+        private static EcdsaKeyPair _mintCntrlKeyPair =
+            new EcdsaKeyPair("0xE83385AF76B2B1997326B567461FB73DD9C27EAB9E1E86D26779F4650C5F2B75".HexToBytes()
+                .ToPrivateKey());
+        private static byte[] _mintCntrlPubKey = CryptoUtils.EncodeCompressed(_mintCntrlKeyPair.PublicKey);
+        private static UInt160 _mintCntrlAdd = Crypto.ComputeAddress(_mintCntrlPubKey).ToUInt160();
+
+        private Money _allowedSupply = Money.Zero;
```

Comment:
This values should be stored in storage,  in other case we will have a zero allowed supply every time we recreate the contract instance

Reply:
added respective methods in `BalanceSnapshot.cs`